docs/fs_group.md

```diff
@@ -0,0 +1,108 @@
+## Manage FSGroup using pod security context and CSI driver spec
+	
+We can manage permission change of volume using fsGroup. This helps non root process to access the volume. CSI driver spec and Pod security context helps us on when to apply permission change using fsGroup.
+
+## External links describing this feature
+- https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/1682-csi-driver-skip-permission
+- https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/695-skip-permission-change
+- https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/csi/csi_mounter.go
+- https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/volume.go
+
+## Implementation details
+Volume ownership and permission is managed by kubelet. To mount CSI volume kubelet calls `NodePublishVolume` implemented by `SP`, after successful mount it tries to apply ownership and permission if required. Every volume in kubernetes like configmap, secret, csi volume implements `Mounter` and `Unmounter` interface. Volume ownership and permission is part of `SetUp` method of CSI Mounter.
+
+```bash
+ _________________________ Kubernetes Node _________________________________
+|   _________ kubelet ___________     _________ CSI Node Driver _________   |
+|  |                             |   |                                   |  |
+|  |  - CSI Mounter              |   |                                   |  |
+|  |    - Task a                 |   |                                   |  |
+|  |    - Task b                 |   |  -  NodePublishVolume             |  |
+|  |    - NodePublishVolume      |   |     (How to mount the volume)     |  |
+|  |    - SetVolumeOwnership     |   |                                   |  |
+|  |    - Task x                 |   |                                   |  |
+|  |    - Task y                 |   |                                   |  |
+|  |_____________________________|   |___________________________________|  |
+|___________________________________________________________________________|
+```
+
+## Configuration
+
+Storage admin can set `FSGroupPolicy` in CSI Driver spec with 3 values.
+- ReadWriteOnceWithFSType - Modify the volume ownership and permissions to the defined `fsGroup` when the `accessMode` is `RWO` and `fsType` is set.
+- None - Mount the volume without attempting to modify volume ownership or permissions.
+- File - Always attempt to apply the defined `fsGroup` to modify volume ownership and permissions regardless of `fsType` or `accessMode`.
+
```

Comment:
can we mention about the default FSGroupPolicy , if we are not explicitly use any policy in CSIDriver spec.